content/en-us/reference/engine/classes/Camera.yaml

```diff
@@ -534,15 +538,28 @@ properties:
+      The dimensions of the device safe area on a Roblox client.
     description: |
-      **ViewportSize** describes the dimensions, in pixels, of the client's
-      viewport.
+      `Class.Camera.ViewportSize` returns the dimensions of the device safe area on the current screen. The device safe area is a rectangle which includes the Roblox top bar area, but does not include any cutouts in the display. The units of `Class.Camera.ViewportSize` are Roblox UI offset units, which may be different from native display pixels.
+
+      <img src="../../../assets/engine-api/enums/ScreenInsets/DeviceSafeInsets.png" width="840"
+      alt="Mobile device screen with cutout showing device safe area." />
+
+      Note: `Class.Camera.ViewportSize` is not equal to the fullscreen area size on displays with cutouts or notches. To obtain the fullscreen area size on all displays, you can query the `AbsoluteSize` property of a `Class.ScreenGui` with `ScreenInsets=None`. See `Enum.ScreenInsets` for a more information about how screen areas are defined.
+      `Class.Camera.ViewportSize` is not equal to the fullscreen size in order to improve cross-platform compatibility and backwards compatibility with legacy experiences.
+
+      <img src="../../../assets/engine-api/enums/ScreenInsets/None.png" width="840"
+      alt="Mobile device screen with cutout showing fullscreen area." />
+
+      Note: `Class.Camera.ViewportSize` is not the actual viewport size the camera uses for rendering. The Camera renders in the fullscreen area. Also the `Class.Camera.FieldOfView` and `Class.Camera.DiagonalFieldOfView` properties are based on the fullscreen area, not `Class.Camera.ViewportSize`.
+      
+      To learn more about how your experience will run on a device with notches/cutouts, please see [this post](https://devforum.roblox.com/t/notched-screen-support-full-release/2074324/1).
 
-      This property ignores the GUI inset applied by the top bar, meaning the
-      center of the screen can be found at precisely at 50% of the viewport in
-      both directions. You can find the position of a `Datatype.Vector3` in the
-      world on the viewport using `Class.Camera:WorldToViewportPoint()`.
+      Only one `Class.Camera` has its `Class.Camera.ViewportSize` updated each frame: the `Class.Camera` currently referred to by `Class.Workspace.CurrentCamera`. The `ViewportSize` of all other `Class.Camera`s in your experience won't be updated. This includes `Class.Camera`s that are used for `Class.ViewportFrame`s.
+
+      The initial value of `Class.Camera.ViewportSize` is `(1, 1)`. The `Class.Workspace.CurrentCamera`'s `Class.Camera.ViewportSize` is updated once each frame during the render prepare step.
+
+      For example, a `Class.ViewportFrame` with `Class.ViewportFrame.CurrentCamera` set to a different `Class.Camera` than the `Class.Workspace.CurrentCamera` will have its `ViewportSize` equal to the initial value of `(1, 1)`.
```

Comment:
My initial instinct is that this might be too much info, or that we can organize it better. Let's process the current comments and suggestions, then see how this fits into the overall description.

Reply:
We do occasionally get DevForum bugs reported by developers about ViewportFrame + Camera, like this one: https://devforum.roblox.com/t/camerascreenpointtoray-doesnt-work-correctly-for-viewportframes/2491032

So I thought by putting all the edgecases in the documentation, we would get fewer reports about this?